### ISSUE TITLE
Ability to get the client version from a Lua method, new "drop the two-handed"

### ISSUE THREAD

**Issue by WibbenZ** [otland/forgottenserver#1584]
Well even if I atleast think the activity has been low here (hopefully this changes soon :p) 

But since Ninja has released this; https://otland.net/threads/8-60-the-forgotten-server-1-2.236489/
We should some sort of function to check if it's the 8.6 or latest version, this way we can do as I have spoken about before; one script for all client version.

The reason this function is needed is mainly because of things like mounts etc that 8.6 does not utilize.
We can either return a boolean function or even better an integer that gives us the version, this way we can check if the version is 10.70 or 10.90 etc

Another thing is the "new" system where a certain game drops the shield or w/e and insted allows the move of a two-handed weapon, this way you save time insted of first having do drop the shield, then move the new weapon to replace your old weapon.

Id like the community input before I or someone else implements these things.

**Comment by raymondtfr** [comment]
I like the idea of checking version, but it'll need much more than a version check (e.g. functions in lua/c++ to disable/not load features of newer versions.) Which I think Ninja is doing, but manually I guess. Anyway, I don't have much experience, so I'm just landing my thoughts xD.

There are a few other new functionalities Tibia has implemented over time that would be really useful in TFS. One of them is the fact that when your character is muted, you'll still be able to cast spells (spell words are not muted.) The "too crowded" (when there are many people on the same tile, you won't be able to do some actions, such as casting spells.) feature, etc.

There is also a new feature I have thought about too (not present in Tibia) which I find very interesting (especially for paladins.) We could display a green message on the screen (like when using potions) of how many Distance weapon (spears, stars, etc) in your SLOT_LEFT_HAND or Ammunition (arrows, bolts) you have in your SLOT_AMMO and display this (by the way, If I'm not mistaken, you can only carry a shield in the shield slot "SLOT_RIGHT_HAND; >".):

* Using one of "x" assassin stars... (btw, for weapons in such slot, it would be a little messy because they have break chance, so it would be spamming the same ammount of remaining stars/spears over and over.)
* Using one of "x" piercing bolts...
* Using one of "x" crystalline arrows...

This would avoid the need of paladins to always keep the equipment window opened to check the remaing ammo (even if we had equipHotkey, it would still be useful.)

**Comment by diath** [comment]
@raymondtfr 
>(btw, for weapons in such slot, it would be a little messy because they have break chance, so it would be spamming the same ammount of remaining stars/spears over and over.)

You could always make it so the message displays only when the weapon breaks, not everytime you shoot.

**Comment by djarek** [comment]
As for the client version querying, it should work like this:
```
function onUse1080(player, item, fromPosition, target, toPosition, isHotkey)
	--do sth for 10.80
	return false
end

function onUse0860(player, item, fromPosition, target, toPosition, isHotkey)
	--do sth for 8.6
	return false
end


onUse = nil
local version = configManager:getClientVersion()

if version == 860 then
	onUse = onUse0860
else if version == 1080 then
	onUse = onUse1080
else
    --print warning or do sth else for unsupported versions
end
```

If there's more common code between the versions then the function object assignment should occur for the inner, version specific functions. This way, the version checks are executed only once, when the script is loaded.

**Comment by WibbenZ** [comment]
@raymondtfr Yes you are correct, but im not talking about source edits, im talking about ex. writing a quest script - that works for both the 8.6 version aswell as the 10.x versions.
So loading scripts has nothing to do with it, since mounts etc are not used in the 8.6 version, all we have to do is add a dummy function for mounts etc etc

Yes that company (Mark has asked us to not mention any names, even in your last PR) has done alot of new things that we also could implement :p 

@djarek The first example would require us to change most function names, what im talking about is what you did below but something like this:

```
local version =  ???
if version == 860 (or version(we call a boolean function to check if it's the latest or not) then
     do 860 stuff
else
     do 10x stuff
end

```

I guess the best thing would be to just add a function that sends the MAX client version and maybe even the MIN version.
That way we can do as you said version == 1055 etc etc

**Comment by djarek** [comment]
@WibbenZ 
I think you misunderstood. The code I've shown isn't 2 examples - it's one example. No function renaming would be necessary (except of course, for scripts that acutally are different between versions). The logic in the local chunk scope is going to be run once and isn't going to impact performance every time the function is executed.

**Comment by WibbenZ** [comment]
Oh sorry diden't read everything haha, nvm it then :p

**Comment by raymondtfr** [comment]
I think he mentioned I should remove any references to that **company** name (haha) in Commit messages, anyway we could always write Tibiae (and we would not infringe anything.)

**Comment by Zbizu** [comment]
@djarek or placing it somewhere on a side of the screen like that: player:say("ammo: " .. amount, TALKTYPE_MONSTER_SAY, true, player, pos)

where pos is something like 0, 0, 0 or 65536, 65536, 15

**Comment by WibbenZ** [comment]
@ranisalt This should not be closed, since we still have the shield / two-handed issue.

**Comment by ranisalt** [comment]
Can you open another issue only with that problem?
